packages/block-library/src/heading/block.json

```diff
@@ -15,7 +15,12 @@
 			"type": "rich-text",
 			"source": "rich-text",
 			"selector": "h1,h2,h3,h4,h5,h6",
-			"role": "content"
+			"role": "content",
+			"control": {
+				"type": "RichText",
+				"label": "Content",
+				"shownByDefault": true
+			}
```

Comment:
I wonder if we can infer a suitable field based on the `type` of the attribute? I did a very basic MVP for that in https://github.com/WordPress/gutenberg/pull/71567.

I definately think your plan to be able to manually specify or override the control component is required though. 

A scenario I've been thinking about is attributes such as `url`. This would typically map to a `<LinkControl>` but that component needs to know the type of entity being searched for. So ultimately we to be able to either:

- provide this data via the block json 
- configure it manually via JSX

I saw you suggested using `index.js` for this which I think might be a good opt-in API for these kinds of edge cases.